docs/contracts/function-modifiers.rst

```diff
@@ -108,13 +108,23 @@ limited to functions of the same library.
 Multiple modifiers are applied to a function by specifying them in a
 whitespace-separated list and are evaluated in the order presented.
 
+Explicit returns from a modifier or function body only leave the current
+modifier or function body. Return variables are assigned and
+control flow continues after the ``_`` in the preceding modifier.
+
 .. warning::
     In an earlier version of Solidity, ``return`` statements in functions
     having modifiers behaved differently.
 
-Explicit returns from a modifier or function body only leave the current
-modifier or function body. Return variables are assigned and
-control flow continues after the "_" in the preceding modifier.
+Modifiers cannot directly access or change the arguments and return values of functions they modify.
+An explicit return from a modifier with ``return;`` does not affect the values returned by the function.
+The only way a modifier can influence the return value is by changing state variables or by not
```

Comment:
Does this work?
```
struct S { uint x; }
modifier m(S memory _s) { _; _s.x = 7; }
function f() m(x) returns (S memory x) {}
```

Reply:
Interesting. It does! The function needs visibility though. Well, looks like I need to make this statement less definitive.

But also I think we should add this to the docs too. It was not obvious to me at all that you can pass return variables to modifiers.

Reply:
I'm not sure we should really document it :)

Reply:
I think it should be documented somewhere. In general, I think the docs often don't go into enough detail. This part of modifier behavior was always unclear to me.

Maybe the problem is that we don't have a detailed language reference and this tries to be both a reference and also serve as a tutorial for new users?

Anyway, I added a mention but without an example so it does not stand out too much.